omegaconf/listconfig.py

```diff
@@ -247,25 +247,11 @@ def __setitem__(self, index: Union[int, slice], value: Any) -> None:
 
     def append(self, item: Any) -> None:
         try:
-            from omegaconf.omegaconf import _maybe_wrap
-
-            index = len(self)
-            self._validate_set(key=index, value=item)
-
-            if isinstance(item, Node):
-                do_deepcopy = not self._get_flag("no_deepcopy_set_nodes")
-                if do_deepcopy:
-                    item = copy.deepcopy(item)
-
-            node = _maybe_wrap(
-                ref_type=self.__dict__["_metadata"].element_type,
-                key=index,
-                value=item,
-                is_optional=_is_optional(item),
-                parent=self,
-            )
-            self.__dict__["_content"].append(node)
+            index = len(self.__dict__["_content"])
+            self.__dict__["_content"].append(None)
+            self._set_item_impl(index, item)
         except Exception as e:
+            del self.__dict__["_content"][index]
```

Comment:
Looks good to me, I would just move the first couple of lines outside of the `try` block, and maybe assign `self.__dict__["_content"]` for readability. Something like (this time I tested, it works :) ):
```python
    def append(self, item: Any) -> None:
        content = self.__dict__["_content"]
        index = len(content)
        content.append(None)
        try:
            self._set_item_impl(index, item)
        except Exception as e:
            del content[index]
            self._format_and_raise(key=index, value=item, cause=e)
            assert False
```

Reply:
done.